search-mapper/src/main/java/org/infinispan/search/mapper/session/SearchSessionBuilder.java

```diff
@@ -0,0 +1,34 @@
+package org.infinispan.search.mapper.session;
+
+import org.hibernate.search.engine.backend.work.execution.DocumentCommitStrategy;
+import org.hibernate.search.engine.backend.work.execution.DocumentRefreshStrategy;
+
+public interface SearchSessionBuilder {
```

Comment:
Sure but this seems to be always part of the same follow up task, that is 
> Use as indexer instead of indexing plan in `QueryInterceptor`

Doesn't it?